test/load-config.js

```diff
@@ -80,10 +97,199 @@ describe(__filename, () => {
     it('throws an error message if an invalid configuration file name is passed', function () {
       const invalidConfig = './fixtures/config/.c8.config.py'
 
-      expect(() => loadConfigFile(invalidConfig)).to.throw(
-        Error,
+      expect(() => loadConfigFile(invalidConfig, baseDI)).to.throw(
+        UnsupportedFileTypeError,
         `Unsupported file type .py while reading file ${invalidConfig}`
       )
     })
+
+    describe('handles file extensions in a case insensitive manner', () => {
+      for (const ext of ['.jSoN', '.JSON']) {
+        it(`reads a JSON file with extension ${ext}`, () => {
+          expect(
+            loadConfigFile(`test${ext}`, {
+              ...baseDI,
+              readFile: () => '{"statements":12}'
+            })
+          ).to.deep.equal({ statements: 12 })
+        })
+      }
+
+      for (const ext of ['.CJS', '.CJs', '.jS', '.JS']) {
+        it(`reads a Common JS file with extension ${ext}`, () => {
+          expect(
+            loadConfigFile(`test${ext}`, {
+              ...baseDI,
+              readJs: () => ({
+                statements: 12
+              })
+            })
+          ).to.deep.equal({ statements: 12 })
+        })
+      }
+
+      for (const ext of ['.Yml', '.YAML']) {
+        it(`reads a YAML file with extension ${ext}`, () => {
+          expect(
+            loadConfigFile(`test${ext}`, {
+              ...baseDI,
+              readFile: () => 'statements: 12'
+            })
+          ).to.deep.equal({ statements: 12 })
+        })
+      }
+    })
+
+    describe('.json', () => {
+      it('throws an error if the JSON file is empty', () => {
+        expect(() =>
+          loadConfigFile('test.json', {
+            ...baseDI,
+            readFile: () => ''
+          })
+        ).to.throw(
+          ConfigParsingError,
+          'Error loading configuration from file test.json: must contain a valid c8 configuration object. Original error: Unexpected end of JSON input'
+        )
+      })
+
+      it('throws an error if the JSON file is invalid JSON', () => {
+        expect(() =>
+          loadConfigFile('test.json', {
+            ...baseDI,
+            readFile: () => 'not valid json'
+          })
+        ).to.throw(
+          ConfigParsingError,
+          'Error loading configuration from file test.json: must contain a valid c8 configuration object. Original error: Unexpected token \'o\', "not valid json" is not valid JSON'
```

Comment:
Different error on Node versions [14.x](https://gist.github.com/mcknasty/7508fa66eb137c53cc0d1f67804df803#file-matrix-test-txt-L253), [16.x](https://gist.github.com/mcknasty/7508fa66eb137c53cc0d1f67804df803#file-matrix-test-txt-L537), [18.x](https://gist.github.com/mcknasty/7508fa66eb137c53cc0d1f67804df803#file-matrix-test-txt-L817) 

[Addressed](https://gist.github.com/mcknasty/80c8c814864241de05987dc3cae5fd54#file-parse-args-patch-diff-L3798-L3809) in Review Branch.

Reply:
🤦‍♂️ Good catch.